File: 09-07/euler4.py
def Euler4():
	def IsPalindrome(s):
		return s[::-1] == s           		# Check if s stepped through backwards is same as s
	largest = 0
	for i in range(100, 1000):			# Loop through all 3 digit numbers
		for j in range(100, 1000):    		# Loop through all 3 digit numbers
			prod = i * j
			if IsPalindrome(str(prod))  	# If the product is a palindrome
        if prod > largest:        			# And the product is greater than out current largest product
			    largest = prod          	# Then the new largest product is prod
	return largest

File: Fall_2018/09-07/euler4/euler4.cpp
#include <iostream>
#include <string>

int euler4(void);
bool palindrome(std::string s);

int main(void) {
  std::cout << euler4() << std::endl;
  return 0;
}

bool palindrome(std::string s) {
  return s == std::string(s.rbegin(), s.rend());
}

int euler4(void) {
  
  int largest = 0;
  
  for (int i = 100; i <= 999; i++) {
    for (int j = 100; j <= 999; j++) {
      
      int prod = i * j;
      if (palindrome(std::to_string(prod))) {
        if (prod > largest) {
          largest = prod;
        }
      }
    }
  }
  return largest;
}
